Apply changes to Shubhamrawat5 / spotify-playlist-downloader

b5i opened the pull request:
Hi !
Just added the metadata accuracy with album search and modify a class.



Can Participate to Hacktoberfest pls ?

b5i committed:
Update class

b5i committed:
Makes work on M1 Macs and on other ARM processors

b5i committed:
Improve metadata accuracy

Shubhamrawat5 commented:
You added album data for query_metadata album but album data is not extracted! It is undefined for all songs

b5i commented:
> You added album data for query_metadata album but album data is not extracted! It is undefined for all songs

Yes I'm working actually on it, the PR is not complete yet.

b5i commented:
> You added album data for query_metadata album but album data is not extracted! It is undefined for all songs

On apple music the update is done

b5i commented:
> You added album data for query_metadata album but album data is not extracted! It is undefined for all songs

For me it's working fine
<img width="842" alt="Capture d’écran 2021-10-22 à 16 45 55" src="https://user-images.githubusercontent.com/44288655/138474865-12f7e815-7181-4647-9a2a-dd939d8eada2.png">
Just fix few line now for the lyrics

b5i committed:
Update package.json

b5i committed:
Update downloader.js

b5i committed:
Add files via upload

b5i committed:
Update searcher.py

b5i commented:
> You added album data for query_metadata album but album data is not extracted! It is undefined for all songs

Strange it's working one time on two...

b5i commented:
> You added album data for query_metadata album but album data is not extracted! It is undefined for all songs

Ok I know, the program does not store on playlist-info.txt the album so it's undefined
I'm fixing that

b5i committed:
Update spotify.js

b5i committed:
Update downloader.js

b5i commented:
Ok all is done and works now ! 💯

b5i commented:
<img width="628" alt="Capture d’écran 2021-10-22 à 17 02 56" src="https://user-images.githubusercontent.com/44288655/138477809-9a798bdf-daf0-4476-9b3c-bd9ac16bd284.png">

b5i committed:
Add files via upload

Shubhamrawat5 commented:
so much changed! 

In spotify.js file, you've changed the query selector value to ".ax_46rB4EOK3hF__KEMO" , which seems pointless and breaking the code.

You put "ax_46rB4EOK3hF__KEMO" points to the number on side of song name
but previously it was "_qbBHRjaGvaZoEZDZ_IY" which point to the heading "3 likes 82 songs, 6 hr 7 min" , here total songs are mentioned so it was taking total song number from there

b5i commented:
Ah yeah ok this thing, it was the only I doubt because first time I run the script it bugs and crash directly so I gladly change it.

b5i committed:
Update spotify.js

b5i commented:
I've just a question :
I had to download 'Breathe' and it downloads 'King' while it shouldn't download and with Apple Music it downloads the right track.
Is it a bug in the script or on the slider.kz side ?

Thanks

b5i committed:
Update README.md

Shubhamrawat5 commented:
Not sure, but i've noticed one thing in slider.kz website that a single character or reorder of words change the whole search result.

If you just put a comma in search query, ordering of songs in result might change

Shubhamrawat5 commented:
Is the lyrics adding function complete?

b5i commented:
> Is the lyrics adding function complete?

Yes, I have fixed all the bugs I faced. I use a lot the Apple Music one so I try to fix as much as possible all the bugs. I will try with a big playlist Spotify and see what's happening.
Apple Music : Fixed. Need to approve the 2 PRs. → I'll test a lot and improve this tool.
Spotify : Fixed too. Need to approve this PR. → I use less Spotify but I'll test it in few days.
Note : I forgot to add on README.md the parameters, I'll improve them tomorrow or day after tomorrow.
Note 2 : Roadmap : 
1. Complete parameters function (-p with playlist, and others things like this).
2. Try to put directly on the file the lyrics (no .lrc file).
3. Enjoy the program.
If you want to contact me without using GitHub my mail address is bollengier.antoine@icloud.com
Thanks for your help !

Shubhamrawat5 commented on downloader.js:
```diff
@@ -5,18 +5,76 @@ const axios = require("axios");
+        lyricsok = false;
+        return;
+    }
+    if (`${stdout}` == 'LYRICS NOT FOUND\n'){
```
Here is a issue, i tried printing stdout value by JSON.stringify(stdout) and it gave 
"LYRICS NOT FOUND\r\n" 

so i think there is \r missing

b5i replied:
<img width="738" alt="Capture d’écran 2021-10-23 à 20 32 50" src="https://user-images.githubusercontent.com/44288655/138567651-aefc946d-803d-46fb-a903-8878151e4191.png">
<img width="358" alt="Capture d’écran 2021-10-23 à 20 32 41" src="https://user-images.githubusercontent.com/44288655/138567653-2c0668f6-f910-4580-a76a-61c6e43b487f.png">
I didn't get this issue

Shubhamrawat5 commented:
Also Lyrics python code is not working properly, It didn't give a single lyrics for me.

I tried to check and found `for result_link in result_links:` is working fine but ` if artist.lower() in lower_title and songname.lower() in lower_title: ` this if condition is getting false everytime so lyrics are not generating.

If you fix this, please update in apple playlist repo too. That would be very helpful

b5i commented:
Got this error : 
EXTRACTING TOTAL SONGS...
spotify-playlist-downloader/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:221
            throw new Error('Evaluation failed: ' + helper_js_1.helper.getExceptionMessage(exceptionDetails));
                  ^

Error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'innerText')
    at __puppeteer_evaluation_script__:4:9
    at ExecutionContext._evaluateInternal (/Users/toine/djo/spotify-playlist-downloader/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:221:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ExecutionContext.evaluate (spotify-playlist-downloader/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:110:16)
    at async getPlaylistList (spotify-playlist-downloader/spotify-playlist-downloader/spotify.js:36:19)
    at async getPlaylistInfo (spotify-playlist-downloader/spotify-playlist-downloader/spotify.js:108:3)
    at async Object.module.exports.getPlaylist (spotify-playlist-downloader/spotify-playlist-downloader/spotify.js:126:3)

Shubhamrawat5 commented:
Try again, i think this happen when your network is slow.

b5i commented:
> Try again, i think this happen when your network is slow.

I've try many many times but that just doesn't work...
I've reinstalled and do all those things but nothing to do...
My speed is 400MBs so I think it's enough to fetch a webpage

b5i commented:
Very strange it worked yesterday.

Shubhamrawat5 commented:
wait, i know why that is happening

Shubhamrawat5 commented:
Put 
` await page.waitForSelector("div[aria-rowindex='2']");`

below `console.log("opened.");` line

Shubhamrawat5 commented:
That is happening because song data takes time to load after page get loaded, so `await page.waitForSelector("div[aria-rowindex='2']");` will hold the code till the first song is shown!

b5i commented:
> That is happening because song data takes time to load after page get loaded, so `await page.waitForSelector("div[aria-rowindex='2']");` will hold the code till the first song is shown!

Sorry, it doesn't work, have you got another div or element that is generated in last time ?

Shubhamrawat5 commented:
It should work, div[aria-rowindex='2'] points to first song.

b5i commented:
To be sure it's like this no ?
<img width="910" alt="Capture d’écran 2021-10-24 à 10 53 29" src="https://user-images.githubusercontent.com/44288655/138587129-73a1df51-ccfa-44c7-9408-d5fb2e6687d8.png">

b5i commented:
> To be sure it's like this no ?
> <img alt="Capture d’écran 2021-10-24 à 10 53 29" width="910" src="https://user-images.githubusercontent.com/44288655/138587129-73a1df51-ccfa-44c7-9408-d5fb2e6687d8.png">

But I've got the same problem

Shubhamrawat5 commented:
Yes, weird it is working fine for me

b5i commented:
> Yes, weird it is working fine for me

Very weird, which version of puppeteer have you got ?

Shubhamrawat5 commented:
"puppeteer": "^10.4.0",

Post the latest error again

b5i commented:
EXTRACTING TOTAL SONGS...
spotify-playlist-downloader/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:221
            throw new Error('Evaluation failed: ' + helper_js_1.helper.getExceptionMessage(exceptionDetails));
                  ^

Error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'innerText')
    at __puppeteer_evaluation_script__:4:9
    at ExecutionContext._evaluateInternal (spotify-playlist-downloader/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:221:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ExecutionContext.evaluate (spotify-playlist-downloader/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:110:16)
    at async getPlaylistList (spotify-playlist-downloader/spotify-playlist-downloader/spotify.js:36:19)
    at async getPlaylistInfo (spotify-playlist-downloader/spotify-playlist-downloader/spotify.js:108:3)
    at async Object.module.exports.getPlaylist (spotify-playlist-downloader/spotify-playlist-downloader/spotify.js:126:3)

b5i commented:
I've just removed a part of the path with my informations

Shubhamrawat5 commented:
Which playlist you're using?

b5i commented:
> Which playlist you're using?

Check yours emails

Shubhamrawat5 commented:
Didn't get any mail yet. It's sr856161@gmail.com

b5i commented:
> Didn't get any mail yet. It's [sr856161@gmail.com](mailto:sr856161@gmail.com)

Sent

Shubhamrawat5 commented:
Okay so Html is different for both playlist, so this is giving error

wait..

b5i commented:
> Okay so Html is different for both playlist, so this is giving error
> 
> wait..

Maybe with the different location

Shubhamrawat5 commented:
Use this

` let totalSongsHtmlInfo = document.querySelectorAll("._qbBHRjaGvaZoEZDZ_IY")[
      document.querySelectorAll("._qbBHRjaGvaZoEZDZ_IY").length - 1
    ].innerText;`

b5i commented:
> Use this
> 
> ` let totalSongsHtmlInfo = document.querySelectorAll("._qbBHRjaGvaZoEZDZ_IY")[ document.querySelectorAll("._qbBHRjaGvaZoEZDZ_IY").length - 1 ].innerText;`

It's better 
Ok now I get this, `throw new Error('Evaluation failed: ' + helper_js_1.helper.getExceptionMessage(exceptionDetails));

Error: Evaluation failed: RangeError: Invalid array length`

b5i committed:
Multi country update

b5i commented:
Fixed !
Just add   ```await page.setExtraHTTPHeaders({
    'Accept-Language': 'en'
});``` at line 116

Shubhamrawat5 committed:
add lyrics not found condition

b5i committed:
Update downloader.js

b5i committed:
Update searcher.py

b5i committed:
Update searcher.py

b5i committed:
Update downloader.js

b5i committed:
Update package.json

b5i committed:
Spotify changes their classes

b5i committed:
Finish class modify

Now it's completely working

Shubhamrawat5 commented:
Pls make another fresh PR.
